features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Tasks/TasksListing.js

```diff
@@ -0,0 +1,123 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+ 'use strict';
+
+ import React, {Component} from 'react'
+ import {Link} from 'react-router-dom'
+ import {Table, Popconfirm, Button, Dropdown, Menu, message} from 'antd';
+
+ import API from '../../data/api'
+
+ class TasksListing extends Component {
+   constructor(props) {
+       super(props);
+       this.state = {
+           workflows: null,
+           selectedRowKeys: []
+       };
+       this.workflow_type = props.match.params.workflow_type;
+       this.onSelectChange = this.onSelectChange.bind(this);
+       this.handleWorkflowComplete = this.handleWorkflowComplete.bind(this);
+   }
+
+
+   handleWorkflowComplete(referenceId, status) {
+       const hideMessage = message.loading("Completing worklow task ...", 0);
+       const api = new API();
+
+       let promised_update = api.completeWorkflow(referenceId, status);
+       promised_update.then(
+           response => {
+             if (response.status !== 200) {
+                 message.error("Something went wrong while updating the workflow task!");
+                 hideMessage();
+                 return;
+             }
+             let workflows = this.state.workflows;
+             for (let workflowsIndex in workflows) {
+                 if (workflows.hasOwnProperty(workflowsIndex) && workflows[workflowsIndex].referenceId === referenceId) {
+                     workflows.splice(workflowsIndex, 1);
+                     break;
+                 }
+             }
+             this.setState({active: false, workflows: workflows});
+             message.success("Updated successfully!");
+
+             hideMessage();
+           }
+       );
+   }
+   componentDidMount() {
+
+       const api = new API();
+       const type = "AM_" + this.workflow_type.toUpperCase() ;
+
+       const promised_workflows = api.getWorkflows(type);
+       /* TODO: Handle catch case , auth errors and ect ~tmkb*/
+       promised_workflows.then(
+           response => {
+              this.setState({workflows: response.obj.list});
+           }
+       );
+   }
+
+   onSelectChange(selectedRowKeys) {
+       this.setState({selectedRowKeys: selectedRowKeys});
+   }
+
+   render() {
+       const {selectedRowKeys, workflows} = this.state;
+       const columns = [{
+           title: 'Description',
+           dataIndex: 'description',
+            width: '60%'
```

Comment:
It is ok to hardcode style information? I understand that we do not currently have a proper way to maintain style details in a component. Until we have such a convention, shall we declare this as a constant with self-explanatory name?